spec.bs

```diff
@@ -494,6 +506,218 @@ The <dfn for=Navigator method>runAdAuction(|config|)</dfn> method steps are:
 
 </div>
 
+<div algorithm="construct a pending fenced frame config">
+
+To <dfn>construct a pending fenced frame config</dfn> given an [=auction config=]
+<var ignore>config</var>:
+1. Return a [=fenced frame config=] with the following members:
+  : [=fenced frame config/mapped url=]
+  :: a struct with the following members:
+    : [=mapped url/value=]
+    :: `"about:blank"`
+
+    : [=mapped url/visibility=]
+    :: "<a for=visibility>`opaque`</a>"
+
+  : [=fenced frame config/container size=]
+  :: TODO
+
+  : [=fenced frame config/content size=]
+  :: null
+
+  : [=fenced frame config/interest group descriptor=]
+  :: a struct with the following members:
+    : [=interest group descriptor/value=]
+    :: a struct with the following members:
+      : [=interest group descriptor/owner=]
+      :: ""
+
+      : [=interest group descriptor/name=]
+      :: ""
+
+    : [=interest group descriptor/visibility=]
+    :: "<a for=visibility>`opaque`</a>"
+
+  : [=fenced frame config/on navigate callback=]
+  :: null
+
+  : [=fenced frame config/effective sandbox flags=]
+  :: a struct with the following members:
+    : [=effective sandbox flags/value=]
+    :: TODO
+
+    : [=effective sandbox flags/visibility=]
+    :: "<a for=visibility>`opaque`</a>"
+
+  : [=fenced frame config/effective enabled permissions=]
+  :: a struct with the following members:
+    : [=effective enabled permissions/value=]
+    :: «"<code>[=attribution-reporting=]</code>", "<code>private-aggregation</code>" (TODO:ref),
+        "<code>shared-storage</code>" (TODO:ref), "<code>shared-storage-select-url</code>"
+        (TODO:ref)»
+
+    : [=effective enabled permissions/visibility=]
+    :: "<a for=visibility>`opaque`</a>"
+
+  : [=fenced frame config/fenced frame reporting metadata=]
+  :: null
+
+  : [=fenced frame config/exfiltration budget metadata=]
+  :: null
+
+  : [=fenced frame config/nested configs=]
+  :: a struct with the following members:
+    : [=nested configs/value=]
+    :: an empty [=list=] «»
+
+    : [=nested configs/visibility=]
+    :: "<a for=visibility>`opaque`</a>"
+
+  : [=fenced frame config/embedder shared storage context=]
+  :: null
+
+</div>
+
+<div algorithm="fill in a pending fenced frame config">
+
+To <dfn>fill in a pending fenced frame config</dfn> given a [=fenced frame config=]
+|pendingConfig|, [=auction config=] |auctionConfig|, and [=leading bid info=] |winningBidInfo|:
+1. Let |winningBid| be |winningBidInfo|'s [=leading bid info/leading bid=].
+1. Set |pendingConfig|'s [=fenced frame config/mapped url=]'s [=mapped url/value=] to
+   |winningBid|'s [=generated bid/ad descriptor=]'s [=ad descriptor/url=].
+1. Let |adSize| be |winningBid|'s [=generated bid/ad descriptor=]'s [=ad descriptor/size=].
+1. If |adSize| is not null:
+   1. Set |pendingConfig|'s [=fenced frame config/content size=] to a struct with the following members:

```

Comment:
nit: link to `struct` and `items` here (instead of members) and below in the rest of this algorithm.